code/modules/mechs/mech_interaction.dm

```diff
@@ -201,6 +201,11 @@
 		if(LAZYLEN(pilots) >= LAZYLEN(body.pilot_positions))
 			to_chat(user, SPAN_WARNING("\The [src] is occupied to capacity."))
 			return FALSE
+	if(ishuman(user) && body?.armor_restrictions)	//wear_suit only exists on humans; only bother with checking if the chassis forbids it
+		var/mob/living/carbon/human/enterer = user
+		if(enterer.wear_suit && enterer.wear_suit.slowdown >= MEDIUM_SLOWDOWN)	//If the user is wearing armor, check if it's too bulky
```

Comment:
Using slowdown will permit some of the best pieces of armor - such as crusader armor - to be worn, meaning it will simply encourage mech users to powergame harder and find these meta armors.

An alternative would be the use of a variable defined on suit level.

Reply:
It might, but I would not be a good judge of saying what armor can or cannot be used. Better to just disallow all of them.